integration/spark-common/src/main/scala/org/apache/carbondata/events/CreateTableEvents.scala

```diff
@@ -27,7 +28,8 @@ import org.apache.carbondata.core.metadata.AbsoluteTableIdentifier
  */
 case class CreateTablePreExecutionEvent(
     sparkSession: SparkSession,
-    identifier: AbsoluteTableIdentifier) extends Event with TableEventInfo
+    identifier: AbsoluteTableIdentifier,
+    tableModel: Option[TableModel]) extends Event with TableEventInfo
```

Comment:
TableModel is a temporary data structure that used in spark integration only, while TableInfo represent the carbon table entity, so we should use TableInfo instead of TableModel in any framework and interface code